components/content/NavigationHeader.tsx

```diff
@@ -7,18 +7,53 @@ import {Title1Black, Title3Black} from 'components/text';
 import React, {useCallback} from 'react';
 import {useSafeAreaInsets} from 'react-native-safe-area-context';
 import {colorLookup} from 'theme';
-import {AvalancheCenterID} from 'types/nationalAvalancheCenter';
+import {AvalancheCenterID, AvalancheCenterWebsites} from 'types/nationalAvalancheCenter';
 
 export const NavigationHeader: React.FunctionComponent<
   NativeStackHeaderProps & {
     center_id: AvalancheCenterID;
     large?: boolean;
   }
 > = ({navigation, route, options, back, center_id, large}) => {
+  let share: boolean = false;
+  let firstOpen: boolean = false;
+  let shareCenterId: AvalancheCenterID = center_id;
+  const shareParams: {share: boolean; share_url: string} = route?.params as {share: boolean; share_url: string};
+
+  if (shareParams.share) {
+    share = true;
+    // if back is false, means the obs screens have not been open yet
+    if (!back) {
+      firstOpen = true;
+      // set back to not be null since we want a shared obs to have a back button
+      back = {title: 'Observations'};
+    }
+
+    const flip = (data: Record<AvalancheCenterID, string>) => Object.fromEntries(Object.entries(data).map(([key, value]) => [value, key]));
```

Comment:
We already have `reverseLookup` in the repo, can you use that?

Reply:
I *think* it is also typed correctly so you should not need `as AvalancheCenterID` on the output.

Reply:
still need it or it complains